x-pack/plugins/alerting/server/task_runner/inject_action_params.test.ts

```diff
@@ -47,6 +55,39 @@ describe('injectActionParams', () => {
     `);
   });
 
+  test('injects viewInKibanaPath and viewInKibanaText when actionTypeId is .email with basePathname and spaceId', () => {
+    const actionParams = {
+      body: {
+        message: 'State: "{{state.value}}", Context: "{{context.value}}"',
+      },
+    };
+    const ruleUrl = {
+      absoluteUrl:
+        'http://localhost:5601/kibana/s/mary/app/management/insightsAndAlerting/triggersActions/rule/1',
+      kibanaBaseUrl: 'http://localhost:5601/kibana',
+      basePathname: '/kibana',
+      spaceIdSegment: '/s/mary',
+      relativePath: '/app/management/insightsAndAlerting/triggersActions/rule/1',
+    };
+    // result.kibanaFooterLink.path should not include /kibana since it is part of kibanaBaseUrl already
+    const result = injectActionParams({
+      actionParams,
+      actionTypeId: '.email',
+      ruleUrl,
+    });
+    expect(result).toMatchInlineSnapshot(`
+      Object {
+        "body": Object {
+          "message": "State: \\"{{state.value}}\\", Context: \\"{{context.value}}\\"",
+        },
+        "kibanaFooterLink": Object {
+          "path": "/s/mary/app/management/insightsAndAlerting/triggersActions/rule/1",
```

Comment:
I added a comment on line 72 of this file. Do you mean to move that comment above this line? Is my comment clear?

Reply:
I agree with you, it is really confusing and it is good that we have a lot of tests for this logic. During implementation, I doubted multiple times if `basePathname` should be included or not 😂 

The better approach, in my opinion, is to pass the absolute URL to the email connector instead of the relative one but I wasn't sure what is the surface area for that change.

Reply:
We can leave it both actually - since I was also confused looking at the test :-)

But feel like it's important to note that in the source as well ...